src/cata_imgui.cpp

```diff
@@ -599,3 +607,60 @@ cataimgui::bounds cataimgui::window::get_bounds()
 {
     return { -1.f, -1.f, -1.f, -1.f };
 }
+
+void cataimgui::window::draw_filter( const input_context &ctxt, bool filtering_active )
+{
+    if( !filter_impl ) {
+        filter_impl = std::make_unique<cataimgui::filter_box_impl>();
+        filter_impl->id = 0;
+        filter_impl->text[0] = '\0';
+    }
+
+    if( !filtering_active ) {
+        action_button( "FILTER", ctxt.get_button_text( "FILTER" ) );
+        ImGui::SameLine();
+        action_button( "RESET_FILTER", ctxt.get_button_text( "RESET_FILTER" ) );
+        ImGui::SameLine();
+    } else {
+        action_button( "QUIT", ctxt.get_button_text( "QUIT", _( "Cancel" ) ) );
+        ImGui::SameLine();
+        action_button( "TEXT.CONFIRM", ctxt.get_button_text( "TEXT.CONFIRM", _( "OK" ) ) );
+        ImGui::SameLine();
+    }
+    ImGui::BeginDisabled( filtering_active );
+    ImGui::InputText( "##FILTERBOX", filter_impl->text,
+                      std::extent_v < decltype( filter_impl->text ) > );
+    ImGui::EndDisabled();
+    if( !filter_impl->id ) {
+        filter_impl->id = GImGui->LastItemData.ID;
+    }
+}
+
+std::string cataimgui::window::get_filter()
+{
+    if( filter_impl ) {
+        return std::string( filter_impl->text );
+    } else {
+        return std::string();
+    }
+}
+
+void cataimgui::window::clear_filter()
+{
+    if( filter_impl && filter_impl->id != 0 ) {
+        ImGuiInputTextState *input_state = ImGui::GetInputTextState( filter_impl->id );
+        if( input_state ) {
+            input_state->ClearText();
+            filter_impl->text[0] = '\0';
+        }
+    }
+}
+
+//void cataimgui::window::set_filter( const std::string &filter )
+//{
+//    // doesnt currently work, relies on API only available in newer ImGUi, because I can't have nice things
+//    //ImGuiInputTextState* input_state = ImGui::GetInputTextState( p_impl->id );
+//    //if( input_state ) {
+//    //    input_state->ReloadUserBufAndSelectAll();
+//    //}
+//}
```

Comment:
Can we just, not commit this and the other commented out functions?

Reply:
done